Xamarin.Forms.Core/TapGestureRecognizer.cs

```diff
@@ -35,6 +35,7 @@ public int NumberOfTapsRequired
 		}
 
 		public event EventHandler Tapped;
+		public event TapEventHandler Tap;
```

Comment:
This is fine because is backward-compatible but I am concerned that it will cause confusion. Just reading the names, it appears that Tapped and Tap are exactly the same. Could we rename `Tap` to something that better indicates the difference with `Tapped`? Example: **LocationTapped**.

Reply:
I am even more confused why Xamarin decided not to provide the tapped location in the first place. `Tap` is supposed to be `Tapped` but better. It also conveys the event order: first `Tap`, then `Tapped`. By using a shorter name, `Tap` can be more attractive than `Tapped` and will attract more usage. `Tapped`, being `Tap` but more useless, can be deprecated in the long run.

Reply:
If you insist on the confusion argument, I can deprecate `Tapped` for you.

Reply:
I think a lot of what you're articulating was taken into account with our gestures specs here

https://github.com/xamarin/Xamarin.Forms/issues/3494
and here
https://github.com/xamarin/Xamarin.Forms/issues/3479

In there you have `GestureStatus` to convey order and when things are canceled plus specific events and commands to hook into

The idea was to also generalize these types of args so it could be used by all the gestures..

Thoughts?

Reply:
If those changes can be added in the next release, then I would be happy to side with you. However, seeing that the linked PR is not only stuck in limbo, but also forever in a state of WIP, I believe that a simpler PR such as this one can be merged more quickly and benefit more users.

Reply:
The linked PR is WIP mainly because it also doesn't match the spec

can we reshape this PR slightly to match those specs?

Just taking on a new API to later be deprecated isn't really desireable

Reply:
What are the changes you are specifically talking about?

Reply:
Position on its own doesn't have enough context. Position to what? 

Which those specs address with
```C#
public class Touch
{
	public int TouchIndex { get; set; }
	public Point ViewPosition { get; set; }
	public Point PagePosition { get; set; }
	public Point ScreenPosition { get; set; }
	public GestureStatusType StatusType { get; set; }
	View Target { get; }
}
```

So my thinking would be to create the gesture args described here
https://github.com/xamarin/Xamarin.Forms/issues/3479

you can just implement a subset of those based on what you've already done

Or possibly just match those properties to your `TapEventArgs`

Or we can figure out why the spec we put together is problematic. Make sure we're happy with the path forward and can reach at least a minimal set of APIs that reshapes those specs or just use a subset of those APIs with this PR

Reply:
If `VisualElement` has `ViewX`, `PageX` and `ScreenX`, then I will agree with the API. However, `VisualElement` has the one and only `X`. Therefore, context is already achieved.